handlers/create.go

```diff
@@ -168,3 +184,113 @@ func buildMachineCreateCmd(machine *client.Machine) ([]string, error) {
 	log.Infof("Cmd slice: %v", cmd)
 	return cmd, nil
 }
+
+func createExtractedConfig(event *events.Event, machine *client.Machine) (string, error) {
+	// We are going to ignore doing anything for VirtualBox given that there is no way you can
+	// use Machine once it has been created.  Virtual is mainly a test-only use case
+	if strings.ToLower(machine.Driver) == "virtualbox" {
+		log.Debug("VirtualBox machine does not need config extracted")
+		return "", nil
+	}
+
+	// We will now zip, base64 encode the machine directory created, and upload this to cattle server.  This can be used to either recover
+	// the machine directory or used by Rancher users for their own local machine setup.
+	log.WithFields(log.Fields{
+		"resourceId": event.ResourceId,
+	}).Info("Creating and uploading extracted machine config")
+
+	// tar.gz the $CATTLE_HOME/machine/<machine-id>/machine/<machine-name> dir
+
+	// Open the source directory and read all the files we want to tar.gz
+	baseDir := filepath.Join(os.Getenv("CATTLE_HOME"), "machine", machine.ExternalId, "machine", "machines")
+	machineDir := filepath.Join(baseDir, machine.Name)
+	dir, err := os.Open(machineDir)
+	if err != nil {
+		return "", err
+	}
+
+	log.WithFields(log.Fields{
+		"machineDir": machineDir,
+	}).Debug("Preparing directory to be tar.gz")
+
+	defer dir.Close()
+
+	// Be able to read the files under this dir
+	files, err := dir.Readdir(0)
+	if err != nil {
+		return "", err
+	}
+
+	// create the tar.gz file
+	destFile := filepath.Join(baseDir, machine.Name+".tar.gz")
+	tarfile, err := os.Create(destFile)
+	if err != nil {
+		return "", err
+	}
+
+	defer tarfile.Close()
+	var fileWriter io.WriteCloser = gzip.NewWriter(tarfile)
+	defer fileWriter.Close()
+
+	tarfileWriter := tar.NewWriter(fileWriter)
+	defer tarfileWriter.Close()
+
+	// Read and add files into <machine-name>.tar.gz
+	for _, fileInfo := range files {
+
+		// For now, we will skip directories.  If we need to zip directories, we need to revist this code
+		if fileInfo.IsDir() {
+			continue
+		}
+
+		file, err := os.Open(filepath.Join(machineDir, fileInfo.Name()))
+		if err != nil {
+			return "", err
+		}
+
+		defer file.Close()
+
+		// prepare the tar header
+		header := new(tar.Header)
+		header.Name = filepath.Join(machine.Name, fileInfo.Name())
+		header.Size = fileInfo.Size()
+		header.Mode = int64(fileInfo.Mode())
+		header.ModTime = fileInfo.ModTime()
+
+		err = tarfileWriter.WriteHeader(header)
+
```

Comment:
Not a big deal, but since you're going to be in the file anywayn can you remove this newline so that the error check is right after the err assignment, with no newlines in between?

...I just prefer that style.